sdk/src/azure/core/az_span.c

```diff
@@ -877,22 +876,15 @@ _az_span_scan_until(az_span span, _az_predicate predicate, int32_t* out_index)
   for (int32_t index = 0; index < az_span_size(span); ++index)
   {
     az_span s = az_span_slice_to_end(span, index);
-    az_result predicate_result = predicate(s);
-    switch (predicate_result)
+    bool predicate_result = predicate(s);
+    if (predicate_result)
     {
-      case AZ_OK:
-      {
-        *out_index = index;
-        return AZ_OK;
-      }
-      case AZ_CONTINUE:
-      {
-        break;
-      }
-      default:
-      {
-        return predicate_result;
-      }
+      *out_index = index;
+      return AZ_OK;
+    }
+    else
+    {
+      continue;
```

Comment:
this else is not required anymore

Reply:
Will fix this, in a subsequent PR, post-preview 4.

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-c/pull/1036